




#### 注：
- SDK表示集成库修改，ALG表示算法修改，CPY表示CPython层修改，无标注表示扩展包功能Python使用层面修改
- 聚类pycluster2x扩展包版本号格式如 ```I.II.IIIL``` , 其中： 
  - ```I.II``` 表示基线版本号，如2.3表示隶属于聚档基线2.3的版本
  - ```III```为数字，表示功能迭代版本号，存在集成、算法版本，扩展包功能、特性等更新时，该版本号会进行迭代（**该类更新可能对效果产生影响的风险，必要时伴随测试报告发布**）
  - ```L```为字母，表示修复迭代版本号，当不存在功能更新，仅包含bug修改，不影响算法效果的魔法数字修改等，该版本号会进行迭代（**该类更新通常没有对效果产生影响的风险**）



## [pycluster2x-2.4.2 Release](https://weci.dahuatech.com/#/pipelineHistoryDetail?pipelineHistoryId=Mzk3NDUwMw%3D%3D)
### 更新内容
- 完善了thread_local扩展信息捕获机制，使用pycluster.detail_info()获取扩展信息
```python
with pycluster.detail_info() as detail:
    pydossier.doc()
    print(detail.as_dict())
```
- 修改了异常获取机制，使扩展包有更多的异常可以被python捕获并进行处理
- 优化了```load_small```，3200万纯人脸档案，读取时内存占用下降超2/3 (>503GB -> 176GB) ，耗时降低超2/3 (712562037µs -> 252976205µs)
- CPY - 优化了离线合档代码
### 测试报告
**注**：在pycluster2x-2.4.2中将```multi_core```中```oc_size```默认值改为2400000，使大oc时任务分组数量变少，送入oc_result_reduce的结果变少，导致送入离线合档的数量变少，导致最终人脸召回下降0.04%，符合预期，可以接受

[pycluster2x-2.4.2-old_process_龙泉驿200-323分区配置.xlsx](uploads/39d3a2f82e51fa20569245178f077247/py242_测试报告.xlsx)

[pycluster2x-2.4.2-multi_core_龙泉驿200-323分区配置.xlsx](uploads/1ac9b4b7b5f5b524ccec42429f404d6c/241a_242_mc_测试报告.xlsx)



## [pycluster2x-2.4.1a Release](https://weci.dahuatech.com/#/pipelineHistoryDetail?pipelineHistoryId=Mzg4ODI0NQ%3D%3D)
### 更新内容
- 更新了流程测试脚本，统一使用```save_small```，```load_small```，并在```load_small```时指定新的cfg配置项
- 修改一个人体离线合档会送人脸模态信息的问题


## [pycluster2x-2.4.1 Release](https://weci.dahuatech.com/#/pipelineHistoryDetail?pipelineHistoryId=Mzg3MzYyNA%3D%3D)
### 新特性
- 增加了直接触发底档中档案合并或离线合档的手段，通过```check_multi(self, labels:'list[list[int]]', mode:'pydossier.fuse_e')```接口触发，如：
``` python
dos.check_multi([[0,1,2]], pydossier.fuse_e.MERGE) #将底档中的0,1,2号人合并
dos.check_multi([[0,1,2], [10,11]], pydossier.fuse_e.JUDGE) #将底档中的0,1,2号人进行多档检查（走人脸离线合档逻辑）, 10, 11 号人进行多档检查（走人脸离线合档逻辑）
```
### 更新内容
- ALG - 更新了人脸、人体算法版本，修改了运行ocs接口人脸结果导致效果下降的问题
- 修改了oc输出格式，type分为FACE, BODY, OTHERS，以区分oc结果来源。OTHERS得到的结果分数为1
- ```get_pid_from_rid```接口调用时，如果rid不存在，原来会输出pycluster.INT_MAX，改为抛出python IndexError异常
- ```get_person_from_pid```接口调用时，如果pid越界，原来会崩溃，改为抛出python IndexError异常
- CPY - 修改了一个扩展包内存泄漏的问题
### 报告
[pycluster2x-2.4.1-龙泉驿200-323分区配置.xlsx](uploads/eef89b10dcceb891aab0e0d6aafd36bc/pycluster2x-2.4.1-龙泉驿200-323分区配置.xlsx)
### 已知问题
- 部分数据集oc接口人脸人体会输出DHIVS_CLUSTER_FUSE_RECOMMEND_ADSORBED，送入离线合档接口人脸会报错


## [pycluster2x-2.4.0 Release (~~弃用版本号pycluster2x-2.3.18~~)](https://weci.dahuatech.com/#/pipelineHistoryDetail?pipelineHistoryId=MzgwNzQyNg%3D%3D)
### 新特性
- ALG\SDK\CPY全面更新至基线版本2.004
- 添加通过已有标签构建档案集的方法```simple_batch```
``` python
from pycluster2x import pycluster, pydossier, pypic

ms0 = pypic.pb_input_csv("./pycluster2x/tests/data0.csv")[0:3]  # 取三条数据测试
dossier0 = pydossier.dossier(pycluster.cluster_hdl([pycluster.FACE], pycluster.default_cfg()))
print(ms0)
# 每个模态的labels和dossier_labels长度需要和输入数据中对应模态图片数相同，即每张图要有对应的标签信息，可以通过 ms0.person_num(pycluster.FACE) 或者 ms0.person_num(pycluster.BODY) 获得相关信息
dossier0.simple_batch(ms0, labels = {pycluster.FACE:[0,0,1], pycluster.BODY:[]}, dossier_labels = {pycluster.FACE:[0,0,0], pycluster.BODY:[]})
```
- 通过```__contains__```判断rid是否在档案中，如：
``` python
if "341523210011902306500220210821161147470860603412" in dossier_0:
    print("YES")
else:
    print("NO") 
```
### 更新内容
- CPY - 修正了档案属性拷贝错位的问题
- SDK - 更新集成库版本，修正了档案属性拷贝错位的问题
- ALG - 更新算法版本
- 修改一些已知问题
